onnxscript/_legacy_ir/protobuilder.py

```diff
@@ -108,6 +108,9 @@ def process_ir_node(self, ir_node: ir.Node, node_proto: onnx.NodeProto) -> onnx.
         for attr in ir_node.attributes.items():
             attr_proto = self.process_attribute(attr)
             node_proto.attribute.append(attr_proto)
+        if len(ir_node.attributes) == 0 and len(ir_node.original_node_proto.attribute) > 0:
+            # ir_node.attributes is never populated.
```

Comment:
Should this be fixed in `irbuilder.py`? But I think node attrs should be covered there, otherwise unittests should have caught it. What is the particular case that the attrs weren't populated?

Reply:
May be ok. Anyway, all of this will disappear after Titai's refactoring. The legacy IR is just a wrapper around a original NodeProto ... see Xavier's TODO comment in file above.

Reply:
Agree no need to update code here since refactor will replace this. I guess I'm more interested in the particular case in case it is an edge case that were not considered, since in general attributes should be populated by irbuilder ...